From 5ecbf413f5d68f8a6731231f157e2c746783dc25 Mon Sep 17 00:00:00 2001
From: Ramachandran Venkataramani <ramavenk@codeaurora.org>
Date: Fri, 20 Apr 2018 08:46:52 -0700
Subject: [PATCH] soc: qcom: pil: Skip status reg read for Quin platform

For Quin platorm the status reg read is not ready
to read at the time of a fatal interrupt.

Change-Id: I5affd0c4aa596319643db5c8380e55df7849bfc9
Signed-off-by: Ramachandran Venkataramani <ramavenk@codeaurora.org>
---
 drivers/soc/qcom/pil-q6v5-mss.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/soc/qcom/pil-q6v5-mss.c b/drivers/soc/qcom/pil-q6v5-mss.c
index 1177cac..803585a 100644
--- a/drivers/soc/qcom/pil-q6v5-mss.c
+++ b/drivers/soc/qcom/pil-q6v5-mss.c
@@ -78,7 +78,11 @@ static void restart_modem(struct modem_data *drv)
 static irqreturn_t modem_err_fatal_intr_handler(int irq, void *dev_id)
 {
 	struct modem_data *drv = subsys_to_drv(dev_id);
+#ifndef CONFIG_MSM_GVM_QUIN
 	u32 nmi_status = readl_relaxed(drv->q6->reg_base + QDSP6SS_NMI_STATUS);
+#else
+	u32 nmi_status = 0;
+#endif
 
 	/* Ignore if we're the one that set the force stop GPIO */
 	if (drv->crash_shutdown)
-- 
1.9.1

